### PR TITLE
VIDEO-3723: Chat Integration Layer - Join - 2nd Review

### PR DESCRIPTION
**Pull request by Alton09**
## Description

This PR is a continuation of #186 that completes the work for this story. It encompasses all the `ChatManagerImpl` changes necessary to successfully connect and disconnect to conversations and manage the state for the UI to later consume. Also note that the `RoomViewModel` connects to the `ChatManager`. This code is to test the ConversationManager with a real token provided by the rtc-video plugin and hasn't been fully tested yet. This is in favor of the `ChatTestActivity` from the initial PR which is why it was removed.

## Validation

* Passes CI with new ChatManagerImpl unit tests
* Validated conversation client connection with a real plugin-rtc token.

## Additional Notes

There will be one more final PR to merge all the work to the `feature/conversations` branch.

## Submission Checklist
 - [x] The source has been evaluated for semantic versioning changes and are reflected in ```versionName``` under `app/build.gradle`
 - [x] The `CHANGELOG.md` reflects any **feature**, **bug fixes**, or **known issues** made in the source code

### CODE REVIEW

**Comment by Alton09** [comment]
Pipeline looks good other than a flaky FTL device, Huawei LEO-BX9. Looking to replace it with another similar stable device.

File: app/src/main/java/com/twilio/video/app/ui/room/RoomViewModel.kt [hunk 1]
```diff
@@ -186,6 +186,7 @@ class RoomViewModel(
                 showConnectingViewState()
             }
             is Connected -> {
+                chatManager.connect(roomEvent.token, roomEvent.room.sid)
```
Comment (kbagchiGWC) [inline app/src/main/java/com/twilio/video/app/ui/room/RoomViewModel.kt:189]: Nice

**Review by kbagchiGWC — APPROVED** [review]
Good work! Please make sure ci pipeline is green before merging.

**Comment by Alton09** [comment]
Thanks @kbagchiGWC ! Pipeline looks good. There's a flaky device on FTL, but the same tests are passing on all other devices. Going to go ahead and merge this one.

**Pull request merged**